### PR TITLE
[x265] Delete the bin directory if the platform is Linux

### PR DESCRIPTION
**Pull request by hellozee**
- What does your PR fix?
Deletes the `bin/` directory in Linux for any type of builds. 

- Which triplets are supported/not supported? Have you updated the CI baseline?
Tested on x64-linux with the following triplet,
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)
set(VCPKG_BUILD_TYPE release)
set(VCPKG_CMAKE_SYSTEM_NAME Linux)
```
No, didn't update CI Baseline.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/x265/portfile.cmake [hunk 1]
```diff
@@ -44,7 +44,7 @@ elseif(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsS
     file(RENAME ${CURRENT_PACKAGES_DIR}/bin/x265.exe ${CURRENT_PACKAGES_DIR}/tools/x265/x265.exe)
 endif()
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (PhoebeHui) [inline ports/x265/portfile.cmake:47]: @hellozee, could you use VCPKG_TARGET_IS_LINUX instead?

Could you also update Version to 3.2-4 in CONTROL file?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@hellozee, thanks for the PR!

File: ports/x265/portfile.cmake (hunk 1 above)
Comment (hellozee) [reply ports/x265/portfile.cmake:47]: Sure

**Review by PhoebeHui — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**